pkg/verifier/cosign/cosign.go

```diff
@@ -275,12 +286,29 @@ func (v *cosignVerifier) verifyInternal(ctx context.Context, subjectReference co
 				extension.IsSuccess = false
 				extension.Err = err.Error()
 			} else {
+				extension.Summary = verificationMessage(bundleVerified, &cosignOpts)
 				hasValidSignature = true
 			}
 			extensionListEntry.Verifications = append(extensionListEntry.Verifications, extension)
 		}
 
-		// TODO: perform keyless verification instead if no keys are found
+		// if no keys are found, perform keyless verification
+		if len(keysMap) == 0 {
+			// verify signature with cosign options + perform bundle verification
+			bundleVerified, err := cosign.VerifyImageSignature(ctx, sig, subjectDescHash, &cosignOpts)
+			extension := cosignExtension{
+				IsSuccess:      true,
+				BundleVerified: bundleVerified,
+			}
+			if err != nil {
+				extension.IsSuccess = false
+				extension.Err = err.Error()
+			} else {
+				extension.Summary = verificationMessage(bundleVerified, &cosignOpts)
```

Comment:
question, for keyless verification. Can customer specify multiple identity and issuer for different scope? if yes, does the summary show which one was being used?

Reply:
Each ratify trust policy `scope` can independently be configured with different set of keyless settings. This includes different values for identity and oidc issuer. Within a single scope, if a user trusts multiple identities and/or issuers, they would use the equivalent `RegExp` field to write an expression that matches multipe. Currently the summary does not output that information on which issuer/identity value/expression was in the trust policy that matched. Should we add it?

Reply:
I've added a new field in the extension that indicates the name of the trust policy that matched for this verification. This will allow users to check the matching trust policy config for settings: 
![image](https://github.com/deislabs/ratify/assets/12980641/3d495346-b33e-489f-b437-c6e889220d28)

Reply:
add doc for updated verifierReport based on trust policy

Reply:
> add doc for updated verifierReport based on trust policy

tracked in docs PR here: https://github.com/deislabs/ratify-web/pull/80#issuecomment-2148019142